packages/@aws-cdk/aws-apigateway/lib/deployment.ts

```diff
@@ -0,0 +1,160 @@
+ * deployments are immutable, as long as the resource's logical ID doesn't
+ * change, the deployment will represent the snapshot in time in which the
+ * resource was created. This means that if you modify the RestApi model (i.e.
+ * add methods or resources), these changes will not be reflected unless a new
+ * deployment resource is created.
+ *
+ * To achieve this behavior, the method `addToLogicalId(data)` can be used to
+ * augment the logical ID generated for the deployment resource such that it
+ * will include arbitrary data. This is done automatically for the
+ * `restApi.latestDeployment` deployment.
+ *
+ * Furthermore, since a deployment does not reference any of the REST API
+ * resources and methods, CloudFormation will likely provision it before these
+ * resources are created, which means that it will represent a "half-baked"
+ * model. Use the `addDependency(dep)` method to circumvent that. This is done
+ * automatically for the `restApi.latestDeployment` deployment.
+ */
+export class Deployment extends cdk.Construct {
+    public readonly deploymentId: DeploymentId;
+    public readonly api: RestApiRef;
+
+    private readonly resource: LatestDeploymentResource;
+
+    constructor(parent: cdk.Construct, id: string, props: DeploymentProps) {
+        super(parent, id);
+
+        this.resource = new LatestDeploymentResource(this, 'Resource', {
+            description: props.description,
+            restApiId: props.api.restApiId,
+        });
+
+        if (props.retainDeployments === undefined || props.retainDeployments) {
+            this.resource.options.deletionPolicy = cdk.DeletionPolicy.Retain;
+        }
+
+        this.api = props.api;
+        this.deploymentId = new DeploymentId(() => this.resource.ref);
+    }
+
+    /**
+     * Adds a dependency for this deployment. Should be called by all resources and methods
+     * so they are provisioned before this Deployment.
+     */
+    public addDependency(dep: cdk.IDependable) {
+        this.resource.addDependency(dep);
+    }
+
+    /**
+     * Adds a component to the hash that determines this Deployment resource's
+     * logical ID.
+     *
+     * This should be called by constructs of the API Gateway model that want to
+     * invalidate the deployment when their settings change. The component will
+     * be resolve()ed during synthesis so tokens are welcome.
+     */
+    public addToLogicalId(data: any) {
+        this.resource.addToLogicalId(data);
+    }
+}
+
+class LatestDeploymentResource extends cloudformation.DeploymentResource {
+    private originalLogicalId: string;
+    private customLogicalId?: string;
+    private hashComponents = new Array<any>();
+
+    constructor(parent: cdk.Construct, id: string, props: cloudformation.DeploymentResourceProps) {
+        super(parent, id, props);
+
+        this.originalLogicalId = this.logicalId;
+
+        Object.defineProperties(this, {
+            logicalId: {
```

Comment:
Since `logicalId` is not a lazy property but rather just a simple value, it cannot be overridden.
I will look into maybe exposing some capability at the Construct level to make this a bit less ugly.

Reply:
This works for me [in the TypeScript playground](http://www.typescriptlang.org/play/#src=class%20Super%20%7B%0D%0A%20%20%20%20public%20readonly%20attr%3A%20string%20%3D%20'hello'%3B%0D%0A%7D%0D%0A%0D%0A%0D%0Aclass%20Sub%20extends%20Super%20%7B%0D%0A%20%20%20%20public%20get%20attr()%3A%20string%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20'fiets'%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aconst%20inst%20%3D%20new%20Sub()%3B%0D%0Aconsole.log(inst.attr)%3B):

```ts
class Super {
    public readonly attr: string = 'hello';
}


class Sub extends Super {
    public get attr(): string {
        return 'fiets';
    }
}

const inst = new Sub();
console.log(inst.attr); // 'fiets'
```

(There might be something more going on so I won't block the review on it--but if you can still make it work I'd prefer that.)

(Honestly, I don't even understand why it works at all)

Reply:
Oh you can totally override "simple properties" alright. Javascript will happily have you do that!

Reply:
okay, I needed to override both setter and getter 👍 

Still hacky, but maybe less ugly.